### PR TITLE
Nodes and tokens fetch from external api

### PR DESCRIPTION
**Pull request by tanghel**
## Reasoning
- In order to avoid hitting rate limits on ES / gateway, we can specify a set of features to activate fetching of nodes and tokens from an external API, so that we don't perform a lot of requests to the downstream resources and hitting 429 on them
  
## Proposed Changes
- Add external API fetch for the nodes and token routes
- Add 2 features in the config file for nodesFetch and tokenFetch with the fields enabled and serviceURL

## How to test
- when tokenFetch is enabled, calling getAllTokens() should make 2 GET calls through apiService at the serviceURL parameter and not call any other function
- when nodesFetch is enabled, calling getAllNodes() should make 2 GET calls through apiService at the serviceURL parameter and not call any other function

### CODE REVIEW

File: config/config.mainnet.yaml [hunk 1]
```diff
@@ -97,7 +97,13 @@ features:
     deadLetterQueueName: 'api-process-nfts-dlq'
   tps:
     enabled: true
-    maxLookBehindNonces: 100  
+    maxLookBehindNonces: 100
+  nodesFetch:
+    enabled: true
+    serviceUrl: 'https://mainnet-api.multiversx.com'
```
Comment (gabrielmatei) [inline config/config.mainnet.yaml:103]: replace with https://api.multiversx.com

File: config/config.mainnet.yaml [hunk 2]
```diff
@@ -97,7 +97,13 @@ features:
...
+  tokensFetch:
+    enabled: true
+    serviceUrl: 'https://mainnet-api.multiversx.com'
```
Comment (gabrielmatei) [inline config/config.mainnet.yaml:106]: replace with https://api.multiversx.com

File: src/endpoints/nodes/node.service.ts [hunk 3]
```diff
@@ -392,6 +398,27 @@ export class NodeService {
     return nodes;
   }
 
+  private async getAllNodesFromApi(): Promise<Node[]> {
+    try {
+      const requestNodesCount = await this.apiService.get(this.apiConfigService.getNodesFetchServiceUrl() + '/nodes/count');
```
Comment (gabrielmatei) [inline src/endpoints/nodes/node.service.ts:403]: ``${this.apiConfigService.getNodesFetchServiceUrl()}/nodes/count``

File: src/endpoints/nodes/node.service.ts [hunk 4]
```diff
@@ -392,6 +398,27 @@ export class NodeService {
...
+      };
+
+      const requestNodes = await this.apiService.get(this.apiConfigService.getNodesFetchServiceUrl() + '/nodes', requestUrlParams);
+      const nodes = requestNodes.data;
+
+      return nodes;
+    } catch (error) {
+      this.logger.error('An unhandled error occurred when getting nodes from API');
+      this.logger.error(error);
+      return [];
```
Comment (gabrielmatei) [inline src/endpoints/nodes/node.service.ts:418]: throw error

File: src/endpoints/tokens/token.service.ts [hunk 5]
```diff
@@ -721,6 +722,10 @@ export class TokenService {
   async getAllTokensRaw(): Promise<TokenDetailed[]> {
     this.logger.log(`Starting to fetch all tokens`);
```
Comment (gabrielmatei) [inline src/endpoints/tokens/token.service.ts:723]: move this log after if

**Review by cfaur09 — APPROVED** [review]

**Pull request merged**